samza-yarn/src/main/java/org/apache/samza/config/YarnConfig.java

```diff
@@ -219,6 +227,10 @@ public String getYarnApplicationModifyAcl() {
     return get(YARN_APPLICATION_MODIFY_ACL, null);
   }
 
+  public String getYarnApplicationType() {
```

Comment:
There is no specific semantics that I would want to pin this config to.
I look at it as a metadata tag that can be set during a deployment of a job.
The use case that I have in mind is to differentiate the deploys between Beam, SQL etc. I imagine others would want to use it for other things.

With regards to my specific use case I plan on populating the ApplicationType in a rewriter.

Reply:
+1, any per-job or cluster-wide controller would want to distinguish between these, since each type API can offer different degrees of insights into workings of a job. 
(low-level: only input, output, high-level, sql: app dag)

Reply:
@rmatharu Samza's controllers shouldn't distinguish between APIs/DSLs for resource management purposes (other than maybe b/w low level vs high level for the reason you mentioned).

@abhishekshivanna `I imagine others would want to use it for other things.`. We generally don't want to distinguish between these from within the core framework. What specifically do you want to do? Can it be done without introducing a new config?

Reply:
Please correct me if i'm wrong.

Another way to determine the information you want is to do the following:

1. Beam, Sql, YARN have their respective ` ApplicationRunner` implementations. Determining which `ApplicationRunner` from the config(`app.runner.class`) can help you determine the type of the application(Beam/Yarn/Sql).
2. Inspecting the `app.class` configuration and determining the sub-type of it(`TaskApplication`, `StreamApplication`) will enable you to determine if you're running a low-level and high-level class. 

Besides the above two points, I'm not sure if (high-level/low-level API, beam/yarn/sql) could be part of *YarnApplicationType*.  Since these are application related properties and no way related to YARN.

Reply:
@prateekm this config is not to be used by the core framework to use in order to distinguish between the types.
Its is meant to have the `Application Type` be populated as part of YARN's APIs, which makes this type queryable for tools that need this information.
![image](https://user-images.githubusercontent.com/1762473/61473365-22970d00-a93b-11e9-966b-8863636b93bc.png)
I still think this should be `yarn.application.type` and not `samza.internal.app.type` because its not meant to be used by the Samza framework.

@shanthoosh hopefully the above comment addresses your question as well.